[Unified search] Improves the current filter/search experience

stratoula opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/129093
Closes https://github.com/elastic/kibana/issues/127244
Closes https://github.com/elastic/kibana/issues/127245
Closes https://github.com/elastic/kibana/issues/117048
Closes https://github.com/elastic/kibana/issues/75289
Closes https://github.com/elastic/kibana/issues/130992

This PR contains all the changes that are related to the unified search phase 1 project. A bunch of things have changed and improved from the design perspective but this phase doesn't add new functionalities. Its goal is to improve the current experience.

![uni3](https://user-images.githubusercontent.com/17003240/164400192-e51b33fe-ce32-478d-8b7c-4396a88c28ba.gif)
![uni4](https://user-images.githubusercontent.com/17003240/164401263-a765ab4d-5799-4e79-89e1-7f735cb1d758.gif)

### What has changed
- The search bar has now another prop to display a dataview picker with additional controls (add new dataview, add runtime fields etc). Applications: Lens, discover
- We now have one menu for the language switcher, filter controls and saved queries management
- The layout has changed significantly when there is no search input (showQueryInput: false). Applicatiions: Controls, Dashboard embedded
<img width="1731" alt="image" src="https://user-images.githubusercontent.com/17003240/164406793-7cceae4a-d970-4691-bddf-1927c3348725.png">

### Storybook
A storybook was added for previewing the SearchBar and its different appearance depending on specific prop values.
`node scripts/storybook --unified_search`

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

stratoula committed:
[Unified search] Moves dataview picker to the search bar (#126560)

* [Unified search] Moves dataview picker to the search bar

* alter texts

* Remove unused file

* [ChangeDataView] Design cleanup

* Fix services mock failure

* Show newly created datavuew on the list

* Keep dataview picker in discover in mobile view

* Cleanup

* Cleanup translations

* Fix some discover FTs

* Fix management FTs

* More test fixes

* Added a dismissible tour

* Pulled the selectabl list into a new component

… for reuse

Called `DataViewsList`. I then changed Lens’ config panel’s own EuiSelectableList to use this new component instead.

*Didn’t do any test updates*

* Fix broken jest test

* Use the same picker component on Discover mobile view

* Apply some CI fixes

* Fix more functional tests

* More FTs fixes

* Close the tour popover for lens tests

* More FTs fixes

* Fix lens FTs

* Using new `styles.ts` pattern for custom styles and allowing for `fullWidth` buttons

* Better tour text and i18n

* Update copy

* No exclamation point

* Cleanup

* Fixes on discover tests

* Fixes on Lens Fts - create runtime fields

* Fixes on edit permission of add field in discover and some FTs fixes

* Further Fts fixes

* More FTs fixes

* Made tour opt-in with `showNewMenuTour`

* Refactor the OSS FTs to change less files

* Further cleanup on the FTs

* Remove unecessary action

* Fix dataview creation bug

* Add a unit test to the new component

* More fixes

* Fix OSS a11y tests

* Adds another unit test for Lens permissions

* Make a change to stabilize the tests

* Clear flyout prop as it is not used anymore

* Deisgn fixes for mobile view

* Address PR comments WIP

* Update the layrpanl dataview list when a new dataview is created

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: cchaos <caroline.horn@elastic.co>
Co-authored-by: Caroline Horn <549577+cchaos@users.noreply.github.com>

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
fix test

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Fix CI

stratoula commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
Fix i18n checks

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
[Unified search] Redesign search bar (#126566)

* [Unified search] Redesign search bar

* Changes to the saved queries design

* Remove `globalQueryBar` from nested under `.kbnTopNavMenu__wrapper`

* [Security Solution] Removing overrides of `.globaQueryBar`

* Cleanup of `.globalQueryBar` styles and variants

* General layout fixes/adjustments

* More general layout fixes and better defaults

* Enable Clear all button if saved query is applied

* Clear should be enabled if searched query is applied but not the save as new

* fix some bugs on enabling and disabling the menu options

* [design] Working on Load Filter set menu

* Indicate selection even before the user clicks apply

* prettify the time text

* Add delete saved qquery confirmation modal

* Some cleaning up

* Fix the manage saved queries link

* Enable load filter set if saved query is created for the first time

* Some fixes on linting and on translations

* Clanup scss

* Make some fixes

* Fix navigatation unit test

* Fix translations check

* Fix rest jest tests

* Fix some functional tests

* Fix checks

* Fixes the appearance of the searchbar components depending on the flags

* Fix OSS tests

* Fixes on tests and query bar logic

* Fix the OSS ally tests

* Fix graph and lens KQL bar display problems, apply saveQuery flag correctly

* Fix more tests

* Fix security solution test

* Undo tweaks for graph

* Better fix for Graph

- Emulate new look, not old
- Uses same context menu panel as the new one
- Provides the same button type (filter button)

* Fix checks and jest test

* Further fixes

* Fix language switcher test

* Hopefully fixes the rest of the solution tests

* Cleanup

* Fixes on maps filtering

* Fix Discover padding

* [Discover] Added New Field button to bottom of fields list

* Simple responsive behavior

* Quick filter updates

* CI fixes

* area-label fix

* 18n translate the add filed btn in discover

* Fix lens functional test

* fix functional tests

* Refactor and add unit test for the add filter button

* Add unit test to the saved management list component

* Stabilize lens test

* Second try to solve the flakiness

* Fix panels depending on the flags

* Fix jest tests

* Fix double menu when only the queryInput is on

* Add storybook for the different layouts of the unified search bar

* Fix ci checks

* Removee hooks comment

* Fix flakiness

* Fix problem with lens dataview change

* Fix checks

* Ensure that the query is submitted

* Fixes oon CI

* Fixes flakiness

* Fix checks CI

* Update texts

* Cleanup

* Fix unskipped funtional test

* Fix FT

* Update src/plugins/unified_search/public/saved_query_form/save_query_form.tsx

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

* Fix eslint

* Fix checks

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: cchaos <caroline.horn@elastic.co>
Co-authored-by: Caroline Horn <549577+cchaos@users.noreply.github.com>
Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

stratoula committed:
Merge with main and resolve conflcits

stratoula committed:
Fix jest

stratoula committed:
Fix jest tests

Commit:
Filter bar / items: Reduce to only necessary wrappers and styles

- Switch to Emotion for styling
- Rename `FilterBadgesWrapper` to `FilterItems`
- Move `FilterItems` and `FilterItem` so own folder `filter_item/`

# Conflicts:
#	src/plugins/unified_search/public/filter_bar/filter_bar.tsx
#	src/plugins/unified_search/public/filter_bar/filter_editor/lib/filter_label.tsx
#	src/plugins/unified_search/public/filter_bar/filter_item/filter_item.tsx
#	src/plugins/unified_search/public/filter_bar/filter_item/filter_items.tsx
#	src/plugins/unified_search/public/filter_bar/filter_view/index.tsx
#	src/plugins/unified_search/public/query_string_input/filter_editor_wrapper.tsx
#	src/plugins/unified_search/public/search_bar/search_bar.tsx

Commit:
Fix styles by removing styled-components

Also added prop `afterQueryBar` to signify that it needs the top margin

# Conflicts:
#	packages/kbn-babel-preset/styled_components_files.js

Commit:
Refactor `SearchBar` and `QueryBarTopRow` a bit

- Added `showSubmitButton` to explicitely hide the submit button (defaults to true)
- SearchBar always renders the `QueryBarTopRow` which now fully controls visibility of the pieces
- Renamed class with new `uni` prefix
- Tried and failed to use Emotion

Commit:
Update usages

# Conflicts:
#	x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx

stratoula committed:
Merge with ES lint commit

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
Stabilize discover security tests

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
[Unified search] Further enhancements (#129877)

* [Unified search] Redesign search bar

* Changes to the saved queries design

* Remove `globalQueryBar` from nested under `.kbnTopNavMenu__wrapper`

* [Security Solution] Removing overrides of `.globaQueryBar`

* Cleanup of `.globalQueryBar` styles and variants

* General layout fixes/adjustments

* More general layout fixes and better defaults

* Enable Clear all button if saved query is applied

* Clear should be enabled if searched query is applied but not the save as new

* fix some bugs on enabling and disabling the menu options

* [design] Working on Load Filter set menu

* Indicate selection even before the user clicks apply

* prettify the time text

* Add delete saved qquery confirmation modal

* Some cleaning up

* Fix the manage saved queries link

* Enable load filter set if saved query is created for the first time

* Some fixes on linting and on translations

* Clanup scss

* Make some fixes

* Fix navigatation unit test

* Fix translations check

* Fix rest jest tests

* Fix some functional tests

* Fix checks

* Fixes the appearance of the searchbar components depending on the flags

* Fix OSS tests

* Fixes on tests and query bar logic

* Fix the OSS ally tests

* Fix graph and lens KQL bar display problems, apply saveQuery flag correctly

* Fix more tests

* Fix security solution test

* Undo tweaks for graph

* Better fix for Graph

- Emulate new look, not old
- Uses same context menu panel as the new one
- Provides the same button type (filter button)

* Fix checks and jest test

* Further fixes

* Fix language switcher test

* Hopefully fixes the rest of the solution tests

* Cleanup

* Fixes on maps filtering

* Fix Discover padding

* [Discover] Added New Field button to bottom of fields list

* Simple responsive behavior

* Quick filter updates

* CI fixes

* area-label fix

* 18n translate the add filed btn in discover

* Fix lens functional test

* fix functional tests

* Refactor and add unit test for the add filter button

* Add unit test to the saved management list component

* Stabilize lens test

* Second try to solve the flakiness

* Fix panels depending on the flags

* Fix jest tests

* Fix double menu when only the queryInput is on

* Add storybook for the different layouts of the unified search bar

* Fix ci checks

* Removee hooks comment

* Fix flakiness

* Fix problem with lens dataview change

* Fix checks

* Ensure that the query is submitted

* Fixes oon CI

* Fixes flakiness

* Fix checks CI

* Update texts

* Cleanup

* Fix unskipped funtional test

* Fix FT

* Fixes filter bar responsiveness

* Add filter button option to the main menu

* Filter form design pass

- Added tooltip around “Add filter” icon button
- Fixed padding
- Stubbed in “Add” vs “Edit” labels

* Query menu icon button tooltip

* ‘Update’ not ‘Edit’ button label

* Apply different strings for adding filter mode vs edit

* Hide Manage SO link if the user donesnt have the permissions

* Fix the wrapping of the badges and datepicker badge

* Minor props naming fix

* Fix checks

* Fix mobile

* Cleanup

* Fix checks

* Faux button group

* Changes on the search input placeholder

* Fix CI

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: cchaos <caroline.horn@elastic.co>
Co-authored-by: Caroline Horn <549577+cchaos@users.noreply.github.com>

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
Fix translations

elasticmachine commented:
Pinging @elastic/kibana-vis-editors @elastic/kibana-vis-editors-external (Team:VisEditors)

stratoula commented:
Thanx Raya! I will look into it!

jbudz reviewed (approved):
storybook/aliases.ts LGTM

michaelolo24 commented:
This is really great work and the interactions look awesome @stratoula! The only thing is in the security solution the filter addition seems permanently disabled. Haven't been able to figure out exactly why yet...
<img width="795" alt="Screen Shot 2022-04-21 at 9 27 44 AM" src="https://user-images.githubusercontent.com/17211684/164476623-6129f6de-ae53-49e1-9b84-687b89df3834.png">

kibanamachine committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Fix security solution add filter bug

stratoula commented:
@michaelolo24 thanx! Fixed! And also thanx for the tip on how to install security data locally. This is super helpful!

yctercero commented:
@stratoula this is awesome! 

On the security solution side - is there any way to filter out a particular index from the data view? I think it would be great for us to shift over to use this fully, but we do need the ability to filter on the data view indices.

cchaos commented:
@yctercero There will be follow up PR's to help solutions integrate the new Data View menu into their own locations. But first we need the general UI updates to be merged. I'll add this as a specific line-item to the [meta issue](https://github.com/elastic/kibana/issues/126095).

stratoula commented:
@yctercero thanx! Do you want to elaborate more on that to understand better the request? Maybe we can create an enhancement issue for the next phases of the unified search project.

stratoula commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into unified-search-1

ghudgins commented:
saw this minor hover issue on the new button group (this a known issue?)

<img width="311" alt="image" src="https://user-images.githubusercontent.com/5465514/164528584-81c7dc9a-5735-44f9-80d1-a8664c5e771e.png">

cchaos commented:
It's known, I was just trying to create the button group quickly to vett through the usability testing. Likely there will still be rounds of code/visual cleanup

michaelolo24 commented:
Thanks @stratoula for the quick fix! Pulled down again, and the functionality is there and working smoothly in both the main security pages as well as timeline from what I have been able to test 👍🏾 💪🏾  Lol @cchaos. The only other stylistic thing which I see is the horizontal line beneath the query bar (which may be known as well). Thanks again all!

<img width="893" alt="image" src="https://user-images.githubusercontent.com/17211684/164533018-9f0f4148-c801-4ef5-9d63-769836ca683e.png">

cchaos commented:
@michaelolo24 Yep, I'm on it! I've got a commit sitting in my local branch. Just trying to find time to switch back to it and push the fix

michaelolo24 reviewed (approved):
Thanks again for the awesome work @stratoula ! LGTM!

Commit:
Fix a test

stratoula committed:
Fix jest test

Commit:
Attaches filter menu when no add button and better dropdown styles

stratoula committed:
Fix lint errors

stratoula committed:
Clearing the query and filters of a saved query should unload the saved query

stratoula commented:
> The experience looks incredible, amazing job!
> 
> I noticed a small issue as I was playing: If you select a filter set and then manually remove the filters individually, the filter pulldown still thinks the filter set is applied and it took me a minute to figure out how to clear it out (using clear all). This seems to happen both for conditions in the search box and filters.

@rayafratkina thanx for that! Fixed!

stratoula committed:
Do not allow to update the loaded saved query if no filters or no query

kertal reviewed (commented):
This is great work 🥳! I like one of the latest changes, that the `Filter set menu` + `Add filter` button are now next to each other. Code LGTM so far started to test and wanted to provide early feedback:

Is the jump of the timepicker into a separate row at a width of about 770px intended? If it is, wouldn't it make sense for it to claim full width like in the `mobile` view
<img width="769" alt="Bildschirmfoto 2022-04-25 um 14 00 32" src="https://user-images.githubusercontent.com/463851/165086098-0b0dafa6-d0db-4da2-a3fe-14ff7ace1a3f.png">

I think when using Graph, a bit of space between the "Language selector" and the query bar might look better?
<img width="618" alt="Graph_-_Elastic" src="https://user-images.githubusercontent.com/463851/165086418-89c159ca-0fb7-4bd7-b655-155ec76a2b61.png">

cchaos commented:
> Is the jump of the timepicker into a separate row at a width of about 770px intended? If it is, wouldn't it make sense for it to claim full width like in the mobile view

@kertal Yes this is intentional, but now we can't have it claim full-width because it's a simple `flex-wrap` so it will only wrap if the length alloted space for the full bar can't support the min-width for each of the individual components. Is it ideal? No. But does it work, yes. We'll continue to iterate in later phases as we introduce the larger concepts. But for now, this was the best way to handle everything that now exists in that top bar.

Commit:
Fix Graph by creating reusable `KibanaFilterButtonGroup`

YulNaumenko reviewed (approved):
LGTM! Excited to see a follow up PR for the solutions, mentioned in the [comment](https://github.com/elastic/kibana/pull/128401#issuecomment-1105548894)

michaelolo24 commented on src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx:
```diff
@@ -58,6 +70,64 @@ export const DiscoverTopNav = ({
+  const editField = useMemo(
+    () =>
+      canEditDataViewField
+        ? async (fieldName?: string, uiAction: 'edit' | 'add' = 'edit') => {
```
Can we make this a bit more functional and move this out into it's own function with `try..catch` and an `abortController`? It'll make it easier to read

stratoula replied:
This code existed in Discover, I just moved it in another file. Exactly same code existed in Lens. I am thinking of cleaning this up a bit but not on this PR

michaelolo24 commented on src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx:
```diff
@@ -58,6 +70,64 @@ export const DiscoverTopNav = ({
+      canEditDataViewField
+        ? async (fieldName?: string, uiAction: 'edit' | 'add' = 'edit') => {
+            if (indexPattern?.id) {
+              const indexPatternInstance = await data.dataViews.get(indexPattern?.id);
```
you shouldn't need the optional chaining here `data.dataViews.get(indexPattern?.id);` since you already check `indexPattern?.id`

stratoula replied:
Done

michaelolo24 commented:
Hey @stratoula , took a look at this again (and it may be due to the conflicts), but noticed the autocomplete is loading at the bottom of the page rather than at the search bar. 

<img width="1689" alt="image" src="https://user-images.githubusercontent.com/17211684/165301304-7609d423-fe2c-4a1e-8d8c-1fac9691c03c.png">

qn895 commented:
Not necessarily an issue since I can reproduce this on main branch as well, but if the filter popup content is open, deleting the filter still keeps the popup open. Note that the popup now shows content for a different filter. Was able to reproduce this in multiple places and it's not a blocker, just wanted to raise it up.

https://user-images.githubusercontent.com/43350163/165337129-49da1ff9-baf9-46b5-a4e1-286c57623a05.mov

Commit:
Use EuiTheme for SearchBar styles

Updated `displayStyle` for Lens & Discover

cchaos commented on src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx:
```diff
@@ -114,6 +196,8 @@ export const DiscoverTopNav = ({
       showSearchBar={true}
       useDefaultBehaviors={true}
+      dataViewPickerComponentProps={dataViewPickerProps}
+      displayStyle="detached"
```
@stratoula I can't figure out why this prop `displayStyle` in Discover's instance is not being passed down to the SearchBar 😕

stratoula replied:
Fixed https://github.com/elastic/kibana/pull/128401/commits/6867bea617e71eb01d4489c6b0146651f60d95cc

cchaos reviewed (approved):
Approving for design knowing that each consuming instance will need to be double-checked after merge.

cchaos commented on x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.scss:
```diff
@@ -13,8 +13,4 @@
     padding: 0;
   }
-  // prevent margin -16 which scrunches the filter bar
-  .globalFilterGroup__wrapper-isVisible {
```
@stratoula Looks like this is stemming from the [same problem I encountered in Discover ](https://github.com/elastic/kibana/pull/128401#discussion_r858871341)where `displayStyle` is not getting passed down when the search bar is implemented via `services.unifiedSearch.ui.SearchBar`.

stratoula replied:
Fixed https://github.com/elastic/kibana/pull/128401/commits/1bb10650587963e5771cd617dd915ba2f30d179c

cchaos replied:
@qn895 It's not great because there's a lot of controls on that single line, but it's aligned now
<img width="1262" alt="Screen Shot 2022-04-26 at 17 46 45 PM" src="https://user-images.githubusercontent.com/549577/165398025-6f3e0574-2bd8-4daa-ab48-544a0b925d0d.png">

stratoula committed:
Fix displayStyle prop problem

stratoula committed:
Correct alignment of data visualizer

stratoula committed:
Super nit comment

qn895 commented on x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.tsx:
```diff
@@ -120,7 +120,7 @@ export const SearchPanel: FC<Props> = ({
     <EuiFlexGroup
       gutterSize="s"
-      alignItems="flexStart"
+      alignItems="center"
```
@stratoula Since we want the the additional filters to flow down without changing the positions of the other controls, I think `center` is not what we want here. I think applying the following patch should resolve this styling issue.

```
Index: x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.tsx	(revision 7074fc14a6951bb0404407f475037db9b517b241)
+++ x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.tsx	(date 1651079563850)
@@ -120,7 +120,6 @@
   return (
     <EuiFlexGroup
       gutterSize="s"
-      alignItems="center"
       data-test-subj="dataVisualizerSearchPanel"
       className={'dvSearchPanel__container'}
       responsive={false}
Index: x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.scss	(revision 7074fc14a6951bb0404407f475037db9b517b241)
+++ x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/search_panel/search_panel.scss	(date 1651079622027)
@@ -3,6 +3,10 @@
   padding: $euiSizeS;
 }
 
+.dvSearchPanel__container {
+  align-items: baseline;
+}
+
 @include euiBreakpoint('xs', 's', 'm', 'l') {
   .dvSearchPanel__container {
     flex-direction: column;
```

stratoula replied:
@qn895 I committed your fix. Thanx for that! How do you feel about Caroline's suggestion?

qn895 replied:
Thanks for adding the fix @stratoula. @cchaos I agree that it's not ideal... But for the purpose of this PR I think it's best to leave the behavior as close to what we currently have on main. This behavior is definitely something we should revisit in 8.3 as a follow up after this PR is merged if that sounds good to ya'll.

kertal reviewed (approved):
Discover & Graph LGTM, tested in the cloud with Safari, Firefox, Chrome. Looks great! 🥳 Looking forward also to use it in our alerting effort, already a  promising draft PR here for exploration: https://github.com/elastic/kibana/pull/130980

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
Data visualizer improvements

stratoula committed:
Apply useMemo in editPermission cont in Discover

stratoula commented on src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx:
```diff
@@ -38,14 +41,23 @@ export const DiscoverTopNav = ({
   const services = useDiscoverServices();
-  const { TopNavMenu } = services.navigation.ui;
+  const { dataViewEditor, navigation, dataViewFieldEditor, data } = services;
+  const editPermission = dataViewFieldEditor.userPermissions.editIndexPattern();
```
@kertal thanx ❤️ Done https://github.com/elastic/kibana/pull/128401/commits/40bb17059b966d6d6d1d4367514dead751c954e7

stratoula committed:
Fix translations checks

stratoula committed:
Fix translations checks

stratoula committed:
Fix filter badge popover bug

Commit:
[Submit button] Changed to EuiSuperUpdateButton

- Quickly adds tooltip content and filled state for more prominence when dirty
- Fixed 18n

cchaos commented:
@stratoula Based on the usability testing, the reduced submit button to a single icon was lowering the prominence of the "Needs updating" state too much. https://github.com/elastic/kibana/pull/128401/commits/c278f67860815011407f0372d21fb8582a908860 fixes that and adds better tooltips (and i18n 😏)

stratoula commented on x-pack/test/functional/apps/maps/feature_controls/maps_security.ts:
```diff
@@ -116,6 +116,7 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
 
       it('allow saving currently loaded query as a copy', async () => {
         await savedQueryManagementComponent.loadSavedQuery('OKJpgs');
+        await queryBar.setQuery('response:404');
```
@nreese we dont allow to copy the same saved filter set now, we allow saving only if the user makes a change. This is already tested above so I will remove this block, u r right :)

stratoula committed:
Fix tests

stratoula committed:
Align the position with the other tolltips and check that functional tests succeed

qn895 commented:
ML changes LGTM 🎉 We'll need to have to do some follow-ups on our end to address the new styling changes but thanks for all your work @stratoula

nreese reviewed (approved):
maps changes LGTM
code review

stratoula committed:
Give some time to the tooltip to disappear

stratoula commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Fix tests

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Merge with main and resolve conflicts

stratoula committed:
Fix CI checks

stratoula committed:
Merge branch 'main' into unified-search-1

spong reviewed (approved):
Checked out, tested locally, and Security Detections code changes LGTM! Thanks for all the enhancements here @stratoula! 🙂

paul-tavares reviewed (approved):
File: 

```
x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/components/search_bar.tsx
```

LGTM 👍

mbondyra reviewed (commented):
Great PR, just a few nitpicks with the code. I'll start testing now :)

stratoula commented on src/plugins/unified_search/public/filter_bar/filter_editor/index.tsx:
```diff
@@ -84,16 +86,31 @@ class FilterEditorUI extends Component<Props, State> {
   }
 
   public render() {
+    const panelTitleAdd = i18n.translate('unifiedSearch.filter.filterEditor.addFilterPopupTitle', {
```
Marta, I think that Caroline added them as constants to be more easily to read. I prefer it that way tbh. Do you mind if I keep them as it is?

mbondyra replied:
no, it's not a big deal - it's just that they are inside of the function so everytime the `render()` function runs, they are being evaluated and the memory is allocated for all for of them even if they are later not used. It's not a real performance issue, just a small thing so I don't mind.

stratoula replied:
I can move them outside of the render function and keep them as constants then!

stratoula replied:
Done https://github.com/elastic/kibana/pull/128401/commits/e86baaf0c30b8db1fd713770e55c149ed195b999

stratoula committed:
Address PR comments

stratoula committed:
Move constants outside render

mbondyra commented:
#### Accessibility testing (only keyboard issues)
1. Dataview popover - You cannot reach extra options for data views when tabbing through navigation

https://user-images.githubusercontent.com/4283304/166666977-1c82e24c-d582-48fc-9d05-331dfaff269d.mp4


2. Filter options popover:
  2.1. When choosing the option that closes the popover via keyboard the focus is lost

https://user-images.githubusercontent.com/4283304/166667052-efa31a74-fe88-4def-ac5c-c8a8e1bb7637.mp4

  2.2. Nit: Also, choosing the option from filter popover some options close the popover, some don’t (changing the language keeps it open, clear filter keeps it open - but I think it’s ok and can be addressed later?) 


3. Add filter popover: Same here - before it was working by focusing the next element, now it looses focus  (screenshot on state on main) after adding a new filter
<img width="692" alt="Screenshot 2022-05-03 at 21 47 08" src="https://user-images.githubusercontent.com/4283304/166666971-b798c3ae-b53d-45c1-9ecb-5c7d33047ad5.png">

mbondyra commented on src/plugins/unified_search/public/query_string_input/language_switcher.tsx:
```diff
@@ -29,132 +25,86 @@ export interface QueryLanguageSwitcherProps {
   anchorPosition?: PopoverAnchorPosition;
   nonKqlMode?: 'lucene' | 'text';
-  nonKqlModeHelpText?: string;
+  isOnMenu?: boolean;
```
it is a nit, no worries <3

stratoula commented:
@mbondyra about the language switcher, it stays open on purpose. About the Clear filter I cant replicate it.
About the third one, I dont think we want to focus on the add filter button now as it is on the top bar

Dosant commented on src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx:
```diff
@@ -308,85 +337,171 @@ export const QueryBarTopRow = React.memo(
+            fill={props.isDirty}
+            data-test-subj="querySubmitButton"
+            // @ts-expect-error Need to fix expecting `children` in EUI
+            toolTipProps={{
```
nit: seems like all other props have stable reference and could have prevented re-render of the SuperUpdateButton. If not create stable reference then better remove React.memo wrapper since it is not working anymore

stratoula commented on src/dev/storybook/aliases.ts:
```diff
@@ -8,6 +8,7 @@
 
 // Please also add new aliases to test/scripts/jenkins_storybook.sh
 export const storybookAliases = {
+  unified_search: 'src/plugins/unified_search/.storybook',
```
Thanx Anton, I didnt know that!

mbondyra commented:
When it comes to the manual testing, I haven't found anything serious. One small issue that we could correct is a glitch of empty string in data view combobox (the combobox selected blue item), when switching the data view in Lens.

https://user-images.githubusercontent.com/4283304/166681208-21978cb4-f6c0-4262-ac47-8a2638b0279a.mp4

stratoula commented on src/plugins/unified_search/public/dataview_picker/change_dataview.styles.ts:
```diff
@@ -0,0 +1,20 @@
+
+export const DATA_VIEW_POPOVER_CONTENT_WIDTH = 280;
+
+export const ChangeDataViewStyles = ({ fullWidth }: { fullWidth?: boolean }) => {
```
Yes I think it is an emotion pattern, @cchaos can give more info here :)

cchaos replied:
Good catch on the naming actually, yes, we can lowercase this to 
```suggestion
export const changeDataViewStyles = ({ fullWidth }: { fullWidth?: boolean }) => {
```

stratoula replied:
I am pushing a commit

stratoula replied:
Done https://github.com/elastic/kibana/pull/128401/commits/eca58851b9c261d923c458bd1465dab5c6be2501

Dosant replied:
Should it also be for those where we actually use `css`? 
for example, `src/plugins/unified_search/public/filter_bar/filter_bar.styles.ts`, `src/plugins/unified_search/public/search_bar/search_bar.styles.ts` 

I think so, since this isn't a React component

stratoula replied:
Correct :) Done!

stratoula committed:
Apply PR comments

stratoula committed:
Fix dataview picker glich

stratoula commented:
@mbondyra glitch is fixed!
About the ally issues, @cchaos can you help? I tried to fix them but I can't find out what is going wrong.

stratoula committed:
Merge with main and resolve conflicta

stratoula committed:
lovercase emotion function

mbondyra commented:
### The problem with DataView menu:
1. EuiContextMenu implements some kind of focus trap - moving all the elements in the menu inside the EuiContextMenu solves the issue (and it should be fine, because items type is `Array<React.ReactElement<any, string | React.JSXElementConstructor<any>>[] | undefined>` 
2. EuiSelectable works as intended after 1. is fixed -> we tab through the elements, but then use arrows up and down to move between items. That's feels odd though for this particular menu, check out the video:

https://user-images.githubusercontent.com/4283304/166726884-e5a0c7c3-c5a1-4c31-a525-8684fd9b60b0.mov

stratoula committed:
Fix a11y problem on dataview picker

stratoula commented:
@mbondyra thanx a lot for looking into it! I think the arrows in the selectable is how the selectable works in EUI. (I see the same behavior in Lens layer dataview picker)
I pushed your solution https://github.com/elastic/kibana/pull/128401/commits/e9f03b18bdad6fc5b89b448476e98417d06f296f ❤️

stratoula committed:
Lowercase rest of emotion style functions

cchaos commented:
@mbondyra & @stratoula I've started a new issue around the return focus issue as a quick investigation shows it's a problem with EuiSelectable within EuiContextMenuPanel, but shouldn't hold up this PR. https://github.com/elastic/kibana/issues/131566

mbondyra commented:
@stratoula for the other problem I've submitted an issue in eui. https://github.com/elastic/eui/issues/5869 It would be great to have it fixed for 8.3, but not blocking this PR. Approved, amazing work 💯

stratoula committed:
Merge branch 'main' into unified-search-1

stratoula committed:
Clean up

dasansol92 reviewed (approved):
Changes on this file `x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/components/search_bar.tsx` LGTM 👍

crob611 reviewed (approved):
Presentation changes LGTM 👍

teresaalvarezsoler commented:
Fwiw, it's a bit difficult to notice the button that users need to click in order to apply the query changes (above all in big screens). Would it be possible to bring the "Refresh" label back again so it's more clear for users that they need to click there? cc/ @ghudgins 

![image](https://user-images.githubusercontent.com/99645649/166946051-26b84bcb-44d8-4e5e-aaee-6fa11555f8f2.png)

cchaos commented:
@teresaalvarezsoler I've moved your comment to an issue we provide discussion and follow up on as to not hold up this PR specifically since we'd like to get the core functionality merged. https://github.com/elastic/kibana/issues/131641

stratoula committed:
Merge branch 'main' into unified-search-1
